app/knight_class.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.equipment = {"armour": [], "weapon": "", "potion": ""}
+        print(f"Knight {name} arrived at the championship!")
+
+    def add_armours(self, armours_list: list) -> None:
+        for item in armours_list:
+            self.equipment["armour"].append(item["part"])
+            self.protection += item["protection"]
```

Comment:
The method `add_armours` assumes that each item in `armours_list` has 'part' and 'protection' keys. Consider adding validation to ensure these keys exist before accessing them to prevent potential KeyError exceptions.